mpf/devices/multiball.py

```diff
@@ -200,9 +242,19 @@ def stop(self):
         self.debug_log("Stopping shoot again of multiball")
         self.shoot_again = False
 
-        # disable shoot again
+        # disable shoot again, grace period, and hurry up
+        self.machine.events.remove_handler(self._grace_period)
+        self.machine.events.remove_handler(self._hurry_up)
         self.machine.events.remove_handler(self._ball_drain_shoot_again)
 
+        self.machine.events.post("multiball_" + self.name + "_grace_period")
```

Comment:
Now we risk to double post the event. One idea would be to add a boolean class property for each of them. Then set those to true on timer start and to false when then handler posted the event. During stop you could check to see if the event has been posted. Additionally, those could become properties for plsceholders (DeviceMonitor on top of the class).

Reply:
Great catch.  I went the placeholder route.  Also added a few other things to only add the handlers if the value was present.  If 0, no point in even posting the event.  So hopefully that will have less events handled, and then only shut it down if true during the stop() method.  If true, it removes the handler, sets it to false, then posts the message.

Thanks for the feedback!  Let me know if you have any other thoughts!

Reply:
Added an extra test to confirm this behavior worked for default case as well.